src/components/shared/AutoComplete/index.tsx

```diff
@@ -51,8 +52,12 @@ export default function AutoComplete({
     toggleOptions((prevStatus) => !prevStatus);
   }, []);
 
-  const onPressOption = useCallback((label: string) => {
-    setInnerValue(label);
+  const onPressOption = useCallback((data: DummyDatum) => {
+    if (data && data.label) {
+      setInnerValue(data.label);
+    }
+    setSelectedData(data);
```

Comment:
it seems `const [selectedData, setSelectedData] = useState<DummyDatum>();` can substitute `const [innerValue, setInnerValue] = useState<string>(value);` completely. Can you manage the related logic?